sdk/core/azure-json/src/main/java/com/azure/json/JsonArray.java

```diff
@@ -0,0 +1,311 @@
+     * after the addition of the new JsonElement object.
+     * TODO: throws error(s) relating to the index in respect to the current state
+     * of the elements ArrayList
+     */
+    public JsonArray addElement(int index, JsonElement element) {
+        // TODO: check for out of bounds
+        this.elements.add(index, element);
+        return this;
+    }
+
+    /**
+     * Sets the index of the JsonArray to the specified JsonElement object,
+     * replacing the existing JsonElement object at this index.
+     *
+     * @param element a JsonElement object representing one of the valid JSON
+     * types: object, array, string, number, boolean, and null
+     * @param index the element index within the JsonArray to replace the current
+     * JsonElement with the newly specified JsonElement object
+     * @return JsonArray object representing the new state of the JsonArray object
+     * after the setting of the new JsonElement object at index.
+     *
+     * TODO: maybe rename this to replaceElement due to the nature of the
+     * ArrayList.set() method
+     * TODO: throws error(s) relating to the index in respect to the current state
+     * of the elements ArrayList
+     */
+    public JsonArray setElement(int index, JsonElement element) {
+        // TODO: check for out of bounds
+        this.elements.set(index, element);
+        return this;
+    }
+
+    /**
+     * Returns a specified JsonElement object from the JsonArray by index
+     *
+     * @param index the index specifying which JsonElement from the JsonArray
+     * object to return.
+     * @return the JsonElement by index from the JsonArray
+     *
+     * TODO: throws error(s) relating to the index in respect to the current state
+     * of the elements ArrayList
+     */
+    public JsonElement getElement(int index) {
+        // TODO: check for index out of bounds
+        return this.elements.get(index);
+    }
+
+    /**
+     * Removes a specified JsonElement object from the JsonArray
+     *
+     * @param index the index specifying which JsonElement from the JsonArray
+     * object to remove.
+     *
+     * TODO: may want to change the return to returning the removed JsonElement
+     * object.
+     *
+     * TODO: throws error(s) relating to the index in respect to the current state
+     * of the elements ArrayList
+     */
+    public JsonElement removeElement(int index) {
+        // TODO: check for index out of bounds
+        this.elements.remove(index);
+        return this;
+    }
+
+    /**
+     * Returns the String representation of the JsonArray object
+     * utilising the JsonWriter to abstract. It passes a StringWriter to the
+     * toWriter method and then returns the resulting String.
+     * @return String representation of the JsonArray object
+     */
+    @Override
+    public String toJson() throws IOException {
+
+        String s = null;
+        try (StringWriter stringOutput = new StringWriter()) {
```

Comment:
For the future once performance testing begins.

`StringWriter` has a lot of synchronization/locking to ensure thread-safety, we don't have to worry about that here. Instead, we could provide a custom internal implementation to be used in scenarios like this which wraps a `StringBuilder` which doesn't have any locking as it doesn't guarantee thread safety.

This could also allow for us to remove the checked IOException here if we wanted to.

Reply:
@JM16450251 this is the one where we pick up the StringBuilderWriter() from core and repackage